src/Transport/ResponseCorrelator.cs

```diff
@@ -18,7 +18,9 @@
 using System;
```

Comment:
I'd remove most of the synchronisation blocks and replace Hashtable.Synchronized(new Hashtable()) with thread safe ConcurrentDictionary.

Reply:
I gave it a try but without this locking, had some group of tests failing.